libs/template/experimental/virtual-scrolling/src/lib/scroll-strategies/autosized-virtual-scroll-strategy.ts

```diff
@@ -0,0 +1,689 @@
+    },
+  ],
+})
+export class AutosizeVirtualScrollStrategy<
+    T,
+    U extends NgIterable<T> = NgIterable<T>
+  >
+  extends RxVirtualScrollStrategy<T, U>
+  implements OnDestroy
+{
+  /**
+   * @description
+   * The amount of items to render upfront in scroll direction
+   */
+  @Input() runwayItems = 20;
+
+  /**
+   * @description
+   * The amount of items to render upfront in reverse scroll direction
+   */
+  @Input() runwayItemsOpposite = 5;
+
+  /**
+   * @description
+   * The default size of the items being rendered. The autosized strategy will assume
+   * this size for items it doesn't know yet. For the smoothest experience,
+   * you provide the mean size of all items being rendered - if possible of course.
+   *
+   * As soon as rxVirtualFor is able to also render actual tombstone items, this
+   * will be the size of a tombstone item being rendered before the actual item
+   * is inserted into its position.
+   */
+  @Input() tombstoneSize = 50;
+
+  /**
+   * @description
+   * Styles that will be applied to a DOM element when being positioned. A useful
+   * example is if you want to implement a css based fade-in animation by setting
+   * the opacity of the rendered item to 1 when getting position.
+   *
+   * @example
+   * \@Component({
+   *   template: `
+   *    <rx-virtual-scroll-viewport
+   *      autosized
+   *      [enterStyles]="{opacity: '1'}"
+   *    ></rx-virtual-scroll-viewport>
+   *   `,
+   *   styles: [`
+   *    .item {
+   *      opacity: 0;
+   *      transition: opacity 125ms linear;
+   *    }
+   *   `]
+   * })
+   * export class VirtualComponent {}
+   */
+  @Input() enterStyles?: Record<keyof CSSStyleDeclaration, string>;
+
+  /**
+   * @description
+   * The autosized strategy uses the native ResizeObserver in order to determine
+   * if an item changed in size to afterwards properly position the views.
+   * You can customize the config passed to the ResizeObserver as well as determine
+   * which result property to use when determining the views size.
+   */
+  @Input() resizeObserverConfig: {
+    options?: ResizeObserverOptions;
+    extractSize?: (entries: ResizeObserverEntry[]) => number;
+  } = {
+    extractSize: (entries) => entries[0].contentRect.height,
+  };
+
+  /** @internal */
+  private viewport: RxVirtualScrollViewport | null = null;
+  /** @internal */
+  private viewRepeater: RxVirtualViewRepeater<T, U> | null = null;
+  /** @internal */
+  private dataDiffer: IterableDiffer<T> | null = null;
+
+  /** @internal */
+  private readonly _contentSize$ = new ReplaySubject<number>(1);
+  /** @internal */
+  readonly contentSize$ = this._contentSize$.asObservable();
+
+  /** @internal */
+  private _contentSize = 0;
+  /** @internal */
+  private set contentSize(size: number) {
+    this._contentSize = size;
+    this._contentSize$.next(size);
+  }
+
+  /** @internal */
+  private readonly _renderedRange$ = new ReplaySubject<ListRange>(1);
```

Comment:
Why not use a `BehaviorSubject`? Then you could have synchronous reads via `this._renderedRange$.value` and wouldn't need the extra `_renderedRange` attribute (same goes for `scrolledIndex`). Or am I missing something?

Reply:
`renderedRange$` is part of the `RxVirtualScrollStrategy` interface, so it's publicly accessible. I didn't want to make a subject public, but just the observable.